### ISSUE TITLE
Change to android contact name/picture does not propagate to linked desktop device

### ISSUE THREAD

**Issue by brad2014** [signalapp/Signal-Android#4798]
To reproduce:
1.  Link signal-desktop to signal-android
2.  Notice that a particular contact has phone number as their name
3.  On android, in Contacts App, edit in correct first name and last name for that phone number (the default account for new contacts is an Exchange account).  Join the signal contact entry to their existing entry which has their picture.
4.  Send a signal message to the edited contact

What I see:
5. After (1), I see an inbox entry for _every_ signal contact (each thread is empty). 
6. In signal-desktop, new message from (4) arrives, but recipient is still listed by phone number, and no photo.

Expected behavior:
7. Inbox entry (thread) for newly edited contact would be listed in Desktop inbox using the contacts correct name and photo.

**Comment by 2-4601** [comment]
I think it just takes some time to propagate (I'm not familiar of how it's done technically). I was going to make an issue about this, however the next day the contact info was synced. Tried all kinds of things to force a sync (such as refreshing Android clients directory) but nothing I did made it update faster.

Would be cool if it was faster though.

I'm using Lollipop's system contact app (which syncs to Google account).

**Comment by 2-4601** [comment]
https://github.com/WhisperSystems/Signal-Desktop/issues/443#issuecomment-161751619
So it seems that changing something in the contacts app should make it sync.

**Comment by weiznich** [comment]
I also tested what @moxie0 suggests there, but this didn' t help.

**Comment by moxie0** [comment]
ah, I was wrong.  It turns out we actually have no way of knowing whether an individual contact's info changed (avatar icon, name, etc...).  So the change will only get propagated if an entirely new Signal contact shows up, at which point we ship everything over.

**Comment by Scindix** [comment]
So there's technically no way to implement this, @moxie0 ? If so it would be at least great to have a button (e.g. in the devices section) that forces an update of all contact data, just like when adding a new one.

**Comment by bobsummerwill** [comment]
I am seeing something which might be a slightly different variant of this issue.

My regular handset is actually a Jolla Phone, not an Android handset, so I ended up using an old Galaxy S3 of mine to get my Signal account started, but am actually only using the Desktop Client after that initial setup.

I connected to a friend of mine in the Desktop Client using his phone number BEFORE adding that number to my Google Contacts, so he shows up as that phone number only, not as his name.

The workaround suggested for this issue talks about touching your Android contacts, which will force a resync, but I'm not using an Android device anymore after that initial setup.

So the Desktop Client doesn't sync from Google Contacts?   It syncs directly from your contacts on an Android device, and pushes those somewhere which the Desktop Client uses?

**Comment by 2-4601** [comment]
> So the Desktop Client doesn't sync from Google Contacts?

That's correct.

> It syncs directly from your contacts on an Android device, and pushes those somewhere which the Desktop Client uses?

Yes. The Android client is supposed to push them to the Desktop client. Except it doesn't work now because of this bug :)

**Comment by bobsummerwill** [comment]
And will never work if you aren't actively running the Android client as well :-)
OK.  Good to know anyway :-)   Thank you, @2-4601.

**Comment by jonathanernst** [comment]
As a workaround you can unlink then relink your desktop

**Comment by bobsummerwill** [comment]
Hey @jonathanernst!

Yeah, so I did that, and it worked.    Sadly, relinking requires you to run the Android client again (to do the QR code scanning).

So I had to dig out my old Galaxy S3 just for that purpose.     I thought that I could get away with only having to do that once, for the original setup.

Since my earlier post, my regular phone has switched from Jolla to Ubuntu Phone (Meizu MX4 Ubuntu Edition), but obviously both still have the same issue ...

That Signal Desktop still has this tie back to you actually having to have an Android phone :-)

But anyway, having gone through that exercise, my contacts are synced again.   For now!
